core/tools.js

```diff
@@ -1627,6 +1627,21 @@
 					yellowgreen: '#9ACD32'
 				},
 
+				_borderStyle: [
+					'none',
+					'hidden',
+					'dotted',
+					'dashed',
+					'solid',
+					'double',
+					'groove',
+					'ridge',
+					'inset',
+					'outset'
+				],
+
+				_widthRegExp: /^(thin|medium|thick|[\+-]?\d+\.?\d*\S*)$/,
```

Comment:
Well, the `[\+-]?\d+\.?\d*\S*` regex works same as `[\+-]?\d+\S*` because `.?` and `\d*` are optional (zero or more occurrences) and `\S*` group already contains them.

It will also match e.g. `12. ` which is invalid value (https://developer.mozilla.org/en-US/docs/Web/CSS/number).

> A length consists of a &lt;number&gt; followed by a unit (px, em, pc, in, mm, …). As with all CSS dimensions, there is no space between the unit literal and the number. The length unit is optional after the number 0.

So for all non `0` values unit is not optional.

I think this regexp should be tweaked to match values more precisely.